lib/srv/db/common/auth.go

```diff
@@ -360,9 +424,52 @@ func (a *dbAuth) GetTLSConfig(ctx context.Context, sessionCtx *Session) (*tls.Co
 			return nil, trace.BadParameter("failed to append CA certificate to the pool")
 		}
 	}
+
+	return tlsConfig, nil
+}
+
+// appendCAToRoot appends CA certificate from session context to provided tlsConfig.
+func appendCAToRoot(tlsConfig *tls.Config, sessionCtx *Session) (*tls.Config, error) {
+	if len(sessionCtx.Database.GetCA()) != 0 {
+		if !tlsConfig.RootCAs.AppendCertsFromPEM([]byte(sessionCtx.Database.GetCA())) {
+			return nil, trace.BadParameter("invalid server CA certificate")
+		}
+	}
 	return tlsConfig, nil
 }
 
+// isCloudDB returns true if database type is AWS RDS/Aurora/Redshift, Azure or Cloud SQL.
+func isCloudDB(sessionCtx *Session) bool {
+	return sessionCtx.Database.IsRDS() || sessionCtx.Database.IsRedshift() ||
+		sessionCtx.Database.IsCloudSQL() || sessionCtx.Database.IsAzure()
+}
```

Comment:
This might be a good method to add to the actual `Database` resource interface e.g. `IsCloudHosted` or something like this (I actually thought about it too).

Reply:
Moved.